hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockManager.java

```diff
@@ -3325,8 +3325,8 @@ private BlockInfo processReportedBlock(
     // Block is on the NN
     LOG.debug("In memory blockUCState = {}", ucState);
 
-    // Ignore replicas already scheduled to be removed from the DN
-    if(invalidateBlocks.contains(dn, block)) {
+    // Ignore replicas already scheduled to be removed from the DN or had been deleted
+    if (storedBlock.isDeleted()) {
```

Comment:
Should we use `storedBlock.isDeleted() || invalidateBlocks.contains(dn, block)`? It is possible that the block has not been deleted, but the replica on this dn is already scheduled for deletion.

Reply:
@zhangshuyan0  Thanks for your comment.  It has been modified。